src/Entity/EntityId.php

```diff
@@ -30,6 +37,68 @@ public function getSerialization() {
 	}
 
 	/**
+	 * Returns an array with 3 elements: the foreign repository name as the first element, the local ID as the last
+	 * element and everything that is in between as the second element.
+	 *
+	 * @param $serialization
+	 * @return array
+	 */
+	public static function splitSerialization( $serialization ) {
+		$parts = explode( ':', $serialization );
+		$localPart = array_pop( $parts );
+		$repoName = array_shift( $parts ) ?: '';
+		$prefixRemainder = implode( ':', $parts );
+
+		return array( $repoName, $prefixRemainder, $localPart );
+	}
+
+	/**
+	 * Concatenates parts of an EntityId serialization with ':'.
+	 *
+	 * @param array $parts
+	 * @return string
+	 */
+	public static function joinSerialization( $parts ) {
+		return implode( ':', array_filter( $parts ) );
+	}
+
+	/**
+	 * Returns the foreign repository name or empty string for local entities.
```

Comment:
Should say that it returns '' if it's a local ID. Should also clarify that for a "chained" foreighn ID, this returns the first part only.